File: 이상욱/BOJ/m2w8/BOJ9613.java
package m2w8;

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class BOJ9613 {
	public static void main(String[] args) throws Exception{
		
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st;
		
		int T = Integer.parseInt(br.readLine());
		
		for(int i = 0; i < T; i++) {
			
			st = new StringTokenizer(br.readLine());
			
			int n = Integer.parseInt(st.nextToken());
			
			int[] arr = new int[n];
			
			for(int j = 0; j < n; j++) {
				arr[j] = Integer.parseInt(st.nextToken());
			}
			
			Long sum = 0L;
			
			for(int j = 0; j < n; j++) {
				for(int k = j+1; k < n; k++) {
					sum += getGCD(arr[j],arr[k]);
				}
			}
			
			System.out.println(sum);
			
		}
		
	}
	
	public static int getGCD(int num1, int num2) {
		
		if(num1 < num2) {
			if(num1 == 0) {
				return num2;
			}else {
				return getGCD(num1, num2 % num1);
			}
		}else {
			if(num2 == 0) {
				return num1;
			}else {
				return getGCD(num2, num1 % num2);
			}
		}
	}
}

File: 이상욱/BOJ/m2w8/BOJ9613.kt
package m2w8

import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.*

fun main() {

    val br = BufferedReader(InputStreamReader(System.`in`))
    var st : StringTokenizer

    val T = br.readLine().toInt()

    repeat(T){
        st = StringTokenizer(br.readLine())
        val n = st.nextToken().toInt()

        val arr = IntArray(n)

        for (i in 0 until n){
            arr[i] = st.nextToken().toInt()
        }

        var sum = 0L
        for(i in 0 until n){
            for(j in i+1 until n){
                sum += getGCD(arr[i],arr[j])
            }
        }
        println(sum)
    }
}

fun getGCD(num1 : Int, num2 : Int) : Int{

    return  if (num1 < num2){
        if(num1 == 0) num2 else getGCD(num1, num2 % num1)
    }else{
        if(num2 == 0) num1 else getGCD(num2, num1 % num2)
    }


}